spark/src/main/scala/org/apache/spark/sql/delta/util/PartitionUtils.scala

```diff
@@ -279,7 +280,9 @@ private[delta] object PartitionUtils {
       validatePartitionColumns: Boolean,
       timeZone: TimeZone,
       dateFormatter: DateFormatter,
-      timestampFormatter: TimestampFormatter): (Option[PartitionValues], Option[Path]) = {
+      timestampFormatter: TimestampFormatter,
+      utcNormalizeTimestamps: Boolean = false,
+      skipCastingToUserType: Boolean = false): (Option[PartitionValues], Option[Path]) = {
```

Comment:
I will try and simplify this API as best I can, it's still backwards compatible in its current state but dislike having to specify 2 more boolean parameters.